### PR TITLE
Add button to join class group chats on Matrix

### PR DESCRIPTION
**Pull request by gabrc52**
Credit: @NumberC (fadif) and @gabrc52 (rgabriel)

Screenshot:

image

This is still a draft PR (pending the questions below, and making live the [web app](https://github.com/gabrc52/class_group_chats) it would redirect to), but I am creating it now to start the conversation.

Blockers:
- [x] I can't recall if I've seen SIPB projects linking to each other. Is there any fundamental opposition to the idea in the first place? I still need to ask on tomorrow's meeting if people are generally fine with it.

Questions:
- [x] Is the `SelectedActivities` component a good place for the button or is there a better place, design-wise? Regardless of the visual placement, would it be better to use a new component instead?
- [x] Does Chakra UI offer a way to make a link (`<a target="_blank" ...>`) that looks exactly like a button? This would be better so that users can preview the link upon hovering, and right click the link to copy its location, or to avoid popup blockers (which do not seem to be triggered for me, but I am unsure if it's true for every single browser configuration)

### CODE REVIEW

**Comment by cjquines** [comment]
> Does Chakra UI offer a way to make a link (<a target="_blank" ...>) that looks exactly like a button?

wrap a button with a link tag?

File: src/components/SelectedActivities.tsx [hunk 1]
```diff
@@ -72,6 +72,27 @@ export function SelectedActivities(props: {
...
+                (activity) => activity instanceof Class
+              ) as Class[];
+              const classNumbers = selectedClasses.map((activity) => activity.number);
+              
+              // Redirect user to the UI to choose group chats on Matrix
+              // Documented in https://github.com/gabrc52/class_group_chats/tree/main/src/routes/import
+              const MATRIX_LINK = 'https://matrix.mit.edu/classes';
+              const referrerQuery = '?via=Hydrant';
+              const numbersQuery = classNumbers.map((number) => `&class=${number}`).join('');
+              const link = MATRIX_LINK + referrerQuery + numbersQuery;
```
Comment (cjquines) [inline src/components/SelectedActivities.tsx:90]:
```ts
const link = `https://matrix.mit.edu/classes?via=Hydrant${
  selectedActivities
    .filter((activity) => activity instanceof Class)
    .map(({ number }) => `&class=${number}`)
    .join('')
}`
```

**Comment by gabrc52** [comment]
> wrap a button with a link tag?

I wrapped it in a `<a>` tag and turned the link as a single expression as suggested.

**Review by psvenk — COMMENTED** [review]
Idea-wise and code-wise, I approve. But UI-wise, I don't know about whether this is the best place to have the button. It feels like something that should be logically grouped along with "Generate .ics file", but that button is admittedly not very prominent (being at the bottom of the page as well as being greyed out until the user mouses over it). Moving that button up to where the group chat button currently is would make the UI too crowded (I would argue that it already is too crowded with such a long button to the right of "+ Activity"), so I would support moving both of them to a third place.

Maybe we could have a new `<Flex>` vertically between the container with the selected classes/activities and the container with the search box, and move both buttons there? That would provide some needed separation while retaining prominence of both the group chat feature and the calendar export feature, though this does raise questions about the position of the "Sign in with Google" button and possibly orphaning the "Preferences" button (the latter of which might just be fine, given that it will presumably be accessed infrequently).

**Comment by cjquines** [comment]
oh, if only there was a [student group in mit that thinks about ui/ux](http://uiux.mit.edu/)

**Comment by gabrc52** [comment]
Thank you!

Two options are to put the button below the activity chips or above:

image

image

I think I'd be inclined to put it above, so that the search and class list are next to each other, but I am not really sure.

The "join group chats" button is associated with the specific class selection and schedule, so it should be somewhat near the list of classes and the schedule. But so are the Google Calendar and ics options.

I agree with the latter comment. I just found out Hydrant has a preferences button. I could experiment with moving the other buttons.

**Comment by gabrc52** [comment]
Thoughts about these UI changes?

Dark:

image

Light:

image

Changes (they are in another branch: https://github.com/gabrc52/hydrant/tree/matrix-and-more-ui-changes):

* Footer is no longer transparent
* The Matrix icon is below schedule switcher and above the list of classes and activities
* The calendar button is next to the Matrix button and now has a calendar icon
* The Matrix and calendar buttons are now teal and blue, respectively
* Sign in with Google icon is gone since the functionality has not yet been reimplemented, but ideally I think there should only be one "Add to calendar" icon which would prompt for which calendar to use (ICS? Google Calendar? Apple calendar, even?)
* The Settings button is for now a "Change theme" button, since the only setting currently available is the theme
* The settings button now has a moon or sun icon, depending on the theme

**Comment by psvenk** [comment]
Ooh, this is nice! I wholeheartedly approve of your redesign and agree with your comments.

**Review by GenericP3rson — COMMENTED** [review]
Looks quite nice to me as well! One comment: thoughts on adding hover text on the Matrix chats button or something of the like to briefly describe what it is? I think it could be useful, but it might come off as clunky—I'm far from a UI expert.

**Comment by gabrc52** [comment]
image

I've merged the UI changes and added a tooltip

**Review by psvenk — APPROVED** [review]

**Pull request merged**